x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -785,24 +804,33 @@ export const RulesList = ({
         <EuiSpacer size="xs" />
         {showRulesList && (
           <>
-            <RulesListFiltersBar
-              inputText={inputText}
-              filters={filters}
-              showActionFilter={showActionFilter}
-              rulesStatusesTotal={rulesStatusesTotal}
-              rulesLastRunOutcomesTotal={rulesLastRunOutcomesTotal}
-              tags={tags}
-              filterOptions={filterOptions}
-              actionTypes={actionTypes}
-              lastUpdate={lastUpdate}
-              showErrors={showErrors}
-              updateFilters={updateFilters}
-              setInputText={setInputText}
-              onClearSelection={onClearSelection}
-              onRefreshRules={refreshRules}
-              onToggleRuleErrors={toggleRuleErrors}
-            />
-            <EuiSpacer size="s" />
+            {showSearchBar ? (
+              <>
+                <RulesListFiltersBar
+                  actionTypes={actionTypes}
+                  filterOptions={filterOptions}
+                  filters={filters}
+                  inputText={inputText}
+                  lastUpdate={lastUpdate}
+                  rulesLastRunOutcomesTotal={rulesLastRunOutcomesTotal}
+                  rulesStatusesTotal={rulesStatusesTotal}
+                  setInputText={setInputText}
+                  showActionFilter={showActionFilter}
+                  showErrors={showErrors}
+                  tags={tags}
+                  updateFilters={updateFilters}
+                  onClearSelection={onClearSelection}
+                  onRefreshRules={refreshRules}
+                  onToggleRuleErrors={toggleRuleErrors}
+                />
+                <EuiSpacer size="s" />
+              </>
+            ) : null}
+
+            {showSearchBar && !isEmpty(filters.ruleParams) ? (
```

Comment:
I guess you want to keep this in order to use the RuleList from the slo details page without showing the search bar?

Reply:
Indeed!